lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -24,18 +24,6 @@ static constexpr llvm::StringRef InputCFile = "revng-input.c";
 static constexpr llvm::StringRef PrimitiveTypeHeader = "primitive-types.h";
 static constexpr llvm::StringRef RawABIPrefix = "raw_";
 
-template<typename T>
-concept HasCustomName = requires(const T &Element) {
-  { Element.CustomName() } -> std::same_as<const model::Identifier &>;
-  { Element.name() } -> std::same_as<model::Identifier>;
-};
-
-template<HasCustomName T>
-static void setCustomName(T &Element, llvm::StringRef NewName) {
-  if (Element.name() != NewName)
-    Element.CustomName() = NewName;
-}
```

Comment:
It boils down to the fact that in your helper you no don't have a set of overloaded `name` methods, but you have `type`, `function`, `field` inside.

Any particular reason for that choice?

Because at the moment it is an obstacle to having an helper like this, which is important to prevent frequent programming errors. So what are the advantages of having `type`, `field`, `function` and friends?

Reply:
Also, ironically none of the `type`, `field`, `function` method return a type, a field or a function. They all return a name.

Reply:
As discussed on mattermost, I dropped this logic and left a TODO.